sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -131,6 +131,7 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       df.explode('letters) {
         case Row(letters: String) => letters.split(" ").map(Tuple1(_)).toSeq
       }
+    assert(!df2.queryExecution.toString.contains("!"))
```

Comment:
ah, I think we should not add `resolved` to `SparkPlan` for this purpose,
how about `assert(df2.queryExecution.executedPlan.missingInput.isEmpty)`?